### PR TITLE
v8: fix build on Linux

### PR DESCRIPTION
**Pull request by danielnachun**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/v8.rb [hunk 1]
```diff
@@ -113,10 +136,57 @@ def install
...
+      exec_prefix=${prefix}
+      libdir=${prefix}/lib
+      includedir=${prefix}/include
+
+      Name: v8
+      Description: V8 JavaScript Engine - Platform library
+      Version: #{version}
+      Libs: -L${libdir} -lv8_libplatform -pthread
+      Cflags: -I${includedir}
+    EOS
```
Comment (SMillerDev) [inline Formula/v8.rb:189]: If upstream doesn't do this, we don't want to do it either. Maybe someone can make a PR for this upstream?

File: Formula/v8.rb [hunk 2]
```diff
@@ -18,11 +18,22 @@ class V8 < Formula
...
+  on_macos do
+    depends_on "llvm" => :build
+    depends_on xcode: ["10.0", :build] # required by v8
+  end
+
+  on_linux do
+    depends_on "pkg-config" => :build
+    depends_on "gcc"
+    depends_on "glib"
+    depends_on "icu4c"
```
Comment (carlocab) [inline Formula/v8.rb:33]: Building against Homebrew ICU is a pain -- can we keep using the bundled one as on macOS?

File: Formula/v8.rb (hunk 1 above)
Comment (carlocab) [reply Formula/v8.rb]: Yes, let's not ship our own `pc` files unless we absolutely really need them.

Comment (danielnachun) [reply Formula/v8.rb]: I removed the `pc` files based on this feedback.  It looks like the easiest way to try to get that added upstream is to open an issue on their bug tracker and find out if this requires more changes to the build system than just adding the files.

File: Formula/v8.rb (hunk 2 above)
Comment (danielnachun) [reply Formula/v8.rb]: Using the bundled ICU definitely helped to simplify the Linux changes!

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**